### PR TITLE
Unit tests update by removing Powermock and Log updates.

### PR DESCRIPTION
**Pull request by cacheung**
Update Consent unit tests by removing Powermock and Changing to Mockito.
Logs update for 2.0 format.

Not include in this PR:
- Functional Test
- Date formatter update to use Core TimeUtil is not final (Waiting for Core change update for Date formatter)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: code/edgeconsent/src/androidTest/java/com/adobe/marketing/mobile/MonitorExtension.java [hunk 1]
```diff
@@ -225,30 +225,6 @@ private void processSharedStateRequest(final Event event) {
 		MobileCore.dispatchResponseEvent(responseEvent, event, null);
```
Comment (prudrabhat) [inline code/edgeconsent/src/androidTest/java/com/adobe/marketing/mobile/MonitorExtension.java:225]: Needs to be updated to use the new api for dispatch. Not sure if this PR includes removing deprecated api usage, but adding this for visibility.

File: code/edgeconsent/src/androidTest/java/com/adobe/marketing/mobile/MonitorExtension.java [hunk 2]
```diff
@@ -16,40 +16,40 @@
...
 
 	@Override
 	protected String getName() {
 		return "MonitorExtension";
 	}
 
+	@Override
+	protected void onRegistered() {
+		MONITOR_EXTENSION_INSTANCE.set(this);
+	}
```
[the remark this answers is not in the document]
Comment (cacheung) [reply code/edgeconsent/src/androidTest/java/com/adobe/marketing/mobile/MonitorExtension.java:46]: This file will be updated in Functional PR.  I was just changed it here to make it build for this PR.

File: code/edgeconsent/src/androidTest/java/com/adobe/marketing/mobile/MonitorExtension.java (hunk 1 above)
Comment (cacheung) [reply code/edgeconsent/src/androidTest/java/com/adobe/marketing/mobile/MonitorExtension.java:225]: Yes, will do that in the Functional PR.  Please ignore this file right now.

File: code/edgeconsent/src/androidTest/java/com/adobe/marketing/mobile/MonitorExtension.java [hunk 3]
```diff
@@ -16,40 +16,40 @@
...
+
+	@Override
+	protected void onRegistered() {
+		MONITOR_EXTENSION_INSTANCE.set(this);
+	}
```
[the remark this answers is not in the document]
Comment (cacheung) [reply code/edgeconsent/src/androidTest/java/com/adobe/marketing/mobile/MonitorExtension.java:51]: This file will be updated in Functional PR.

File: code/edgeconsent/src/androidTest/java/com/adobe/marketing/mobile/edge/consent/ConsentBootUpTests.java [hunk 4]
```diff
@@ -262,6 +251,10 @@ private void initExtensionWithPersistedDataAndDefaults(
 			MobileCore.updateConfiguration(config);
 		}
 
+		List<Class<? extends Extension>> extensions = new ArrayList<>();
```
[the remark this answers is not in the document]
Comment (cacheung) [reply code/edgeconsent/src/androidTest/java/com/adobe/marketing/mobile/edge/consent/ConsentBootUpTests.java:254]: I will take care of this file in the Functional Test PR.

**Comment by cacheung** [comment]
Thanks @timkimadobe and @prudrabhat for reviewing the PR.  It has been updated based on the comments.  
@prudrabhat regarding to the MonitorExtensions, I will have the file updated in Functional test PR.

File: code/edgeconsent/src/test/java/com/adobe/marketing/mobile/edge/consent/ConsentTest.java [hunk 5]
```diff
@@ -68,158 +74,195 @@ public void test_extensionVersionAPI() {
...
+	}
 
-		// The consent extension should register with core
-		PowerMockito.verifyStatic(MobileCore.class, Mockito.times(1));
-		MobileCore.registerExtension(ArgumentMatchers.eq(ConsentExtension.class), callbackCaptor.capture());
+	// ========================================================================================
+	// publicExtensionConstants
+	// ========================================================================================
+	@Test
+	public void test_publicExtensionConstants() throws InterruptedException {
```
Comment (prudrabhat) [inline code/edgeconsent/src/test/java/com/adobe/marketing/mobile/edge/consent/ConsentTest.java:101]: nit: IMO this test is more relevant in functional tests. Latches and thread waits make unit tests flaky.

File: code/edgeconsent/src/test/java/com/adobe/marketing/mobile/edge/consent/ConsentTest.java [hunk 6]
```diff
@@ -68,158 +74,195 @@ public void test_extensionVersionAPI() {
...
+			// verify: not exception when error callback was called
+			callbackCaptor.getValue().error(ExtensionError.UNEXPECTED_ERROR);
+		}
+	}
 
-		// The consent extension should register with core
-		PowerMockito.verifyStatic(MobileCore.class, Mockito.times(1));
-		MobileCore.registerExtension(ArgumentMatchers.eq(ConsentExtension.class), callbackCaptor.capture());
+	// ========================================================================================
+	// publicExtensionConstants
```
Comment (prudrabhat) [inline code/edgeconsent/src/test/java/com/adobe/marketing/mobile/edge/consent/ConsentTest.java:98]: nit: space.

**Review by prudrabhat — APPROVED** [review]

File: code/edgeconsent/src/test/java/com/adobe/marketing/mobile/edge/consent/ConsentTest.java [hunk 7]
```diff
@@ -69,157 +66,197 @@ public void test_extensionVersionAPI() {
...
+	// ========================================================================================
+	@Test
+	public void test_publicExtensionConstants() {
+		assertEquals(ConsentExtension.class, Consent.EXTENSION);
+		List<Class<? extends Extension>> extensions = new ArrayList<>();
+		extensions.add(Consent.EXTENSION);
+		// should not throw exceptions
+		MobileCore.registerExtensions(extensions, null);
```
[the remark this answers is not in the document]
Comment (cacheung) [reply code/edgeconsent/src/test/java/com/adobe/marketing/mobile/edge/consent/ConsentTest.java:104]: After reading Prashanth's comment I am keeping the test as is.  The registerExtension with Consent is tested in Functional test.

File: code/edgeconsent/src/androidTest/java/com/adobe/marketing/mobile/MonitorExtension.java [hunk 8]
```diff
@@ -179,7 +179,7 @@ private void processXDMSharedStateRequest(final Event event) {
 			return;
 		}
 
-		EventData sharedState = getApi().getXDMSharedEventState(stateOwner, event);
+		EventData sharedState = getApi().XDMShareState(stateOwner, event);
```
[the remark this answers is not in the document]
Comment (cacheung) [reply code/edgeconsent/src/androidTest/java/com/adobe/marketing/mobile/MonitorExtension.java:182]: This file is changed in Functional test PR. We can review there.

File: code/edgeconsent/src/androidTest/java/com/adobe/marketing/mobile/MonitorExtension.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (addb) [reply code/edgeconsent/src/androidTest/java/com/adobe/marketing/mobile/MonitorExtension.java:46]: Can you add TODOs for all the files that need to be updated.

File: code/edgeconsent/src/androidTest/java/com/adobe/marketing/mobile/edge/consent/ConsentBootUpTests.java [hunk 9]
```diff
@@ -262,6 +251,10 @@ private void initExtensionWithPersistedDataAndDefaults(
...
+		//extensions.add(Consent.);
+		//extensions.add(MonitorExtension.class);
+		//MobileCore.registerExtensions(extensions, o -> latch.countDown());
```
Comment (addb) [inline code/edgeconsent/src/androidTest/java/com/adobe/marketing/mobile/edge/consent/ConsentBootUpTests.java:257]: Let's remove the comments L255, 256, 257

**Review by addb — APPROVED** [review]
Add todos in code for the changes that will be done in the functional test PR. Also added a comment to clear not required commented out code. Apart from that it looks good!

**Comment by cacheung** [comment]
> Add todos in code for the changes that will be done in the functional test PR. Also added a comment to clear not required commented out code. Apart from that it looks good!

Add the TODOs.
The commented out code needs to be there for avoiding the build error.  Add Todo to explain that.

**Pull request merged**